instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/logging/OpenTelemetryAppenderAutoConfiguration.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.logging;
+
+import io.opentelemetry.api.OpenTelemetry;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@SuppressWarnings("OtelPrivateConstructorForUtilityClass")
+@ConditionalOnBean(OpenTelemetry.class)
+public class OpenTelemetryAppenderAutoConfiguration {
+
+  @Configuration
+  @ConditionalOnProperty(
+      prefix = "otel.springboot.log4j-appender",
+      name = "enabled",
+      matchIfMissing = true)
+  @ConditionalOnClass({
+    io.opentelemetry.instrumentation.log4j.appender.v2_17.OpenTelemetryAppender.class
+  })
+  static class Log4jAppenderConfig {
+
+    @Bean
+    ApplicationListener<ApplicationReadyEvent> log4jOtelAppenderInitializer(
+        OpenTelemetry openTelemetry) {
+      return event -> {
+        io.opentelemetry.instrumentation.log4j.appender.v2_17.OpenTelemetryAppender.install(
```

Comment:
what do you think about adding the otel appender if it doesn't exist? One thing less for the user to do.

Here's how I did it: https://github.com/grafana/grafana-opentelemetry-starter/blob/main/src/main/java/com/grafana/opentelemetry/Log4jConfig.java

Reply:
It would be great. We would have to provide a way to customize the OTel automatically added. Today, [users can do this with an XML configuration](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/b3932c7089dc07b815063ff4ae78db11fa8b3a8a/instrumentation/logback/logback-appender-1.0/library/src/test/resources/logback-test.xml#L13) in their Spring Boot projects. It could be done for example by providing a Spring bean of the OTel appender automatically added. The user could then customize this bean with a Spring post-processor.
The automatic addition of the OTel appenders could perhaps be discussed in an issue and not be implemented in this PR?

Reply:
> We would have to provide a way to customize the OTel automatically added

Users can still add the appender (using XML) if they want to have some special settings.
My proposal is to add the appender only if no otel appender is found.

> separate issue

sure, I don't mind :smile:

Reply:
I have created an issue for tracking and discussions: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8920